### ISSUE TITLE
ArgumentParser PyLance Issue

### ISSUE THREAD

**Issue by tungalbert99** [SupImDos/pydantic-argparse#22]
Pylance is reporting the argument parser is not properly exported from the module, causing it to fail checks.

```
"ArgumentParser" is not exported from module "pydantic_argparse" Pylance[reportPrivateImportUsage](https://github.com/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md#diagnostic-severity-rules)
```

**Comment by SupImDos** [comment]
Hi @tungalbert99

Thanks for raising the issue and the related PR in #23

Interesting, I guess I never noticed this before. I don't normally have the Pylance analysis enabled in my IDE. But, it looks like you're right, Pylance has a ruleset for what it considers a public/private export.

Just for my own notes, see the following:
* [Pylance rules](https://github.com/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md#diagnostic-severity-rules) (`reportPrivateImportUsage`)
* [Pyright re-export rules](https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#library-interface)

I guess this also raises the issue of this package explicitly defining its public interface.

The following code snippet currently raises `21` Pylance errors:

```python
import pydantic_argparse

# Package Root
pydantic_argparse.ArgumentParser
pydantic_argparse.__title__
pydantic_argparse.__description__
pydantic_argparse.__version__
pydantic_argparse.__author__
pydantic_argparse.__license__

# Argparse Subpackage
pydantic_argparse.argparse
pydantic_argparse.argparse.ArgumentParser

# Parsers Subpackage
pydantic_argparse.parsers
pydantic_argparse.parsers.boolean
pydantic_argparse.parsers.command
pydantic_argparse.parsers.container
pydantic_argparse.parsers.enum
pydantic_argparse.parsers.literal
pydantic_argparse.parsers.mapping
pydantic_argparse.parsers.standard

# Utils Subpackage
pydantic_argparse.utils
pydantic_argparse.utils.argument_name
pydantic_argparse.utils.argument_description
pydantic_argparse.utils.namespace_to_dict
pydantic_argparse.utils.type_caster
pydantic_argparse.utils.is_field_a
```

```txt
"ArgumentParser" is not exported from module "pydantic_argparse"  Pylance(reportPrivateImportUsage) [Ln 5, Col 19]
"__title__" is not exported from module "pydantic_argparse"  Pylance(reportPrivateImportUsage) [Ln 6, Col 19]
"__description__" is not exported from module "pydantic_argparse"  Pylance(reportPrivateImportUsage) [Ln 7, Col 19]
"__version__" is not exported from module "pydantic_argparse"  Pylance(reportPrivateImportUsage) [Ln 8, Col 19]
"__author__" is not exported from module "pydantic_argparse"  Pylance(reportPrivateImportUsage) [Ln 9, Col 19]
"__license__" is not exported from module "pydantic_argparse"  Pylance(reportPrivateImportUsage) [Ln 10, Col 19]
"ArgumentParser" is not exported from module "pydantic_argparse.argparse"  Pylance(reportPrivateImportUsage) [Ln 14, Col 28]
"parsers" is not a known member of module  Pylance(reportGeneralTypeIssues) [Ln 17, Col 19]
"parsers" is not a known member of module  Pylance(reportGeneralTypeIssues) [Ln 18, Col 19]
"parsers" is not a known member of module  Pylance(reportGeneralTypeIssues) [Ln 19, Col 19]
"parsers" is not a known member of module  Pylance(reportGeneralTypeIssues) [Ln 20, Col 19]
"parsers" is not a known member of module  Pylance(reportGeneralTypeIssues) [Ln 21, Col 19]
"parsers" is not a known member of module  Pylance(reportGeneralTypeIssues) [Ln 22, Col 19]
"parsers" is not a known member of module  Pylance(reportGeneralTypeIssues) [Ln 23, Col 19]
"parsers" is not a known member of module  Pylance(reportGeneralTypeIssues) [Ln 24, Col 19]
"utils" is not a known member of module  Pylance(reportGeneralTypeIssues) [Ln 27, Col 19]
"utils" is not a known member of module  Pylance(reportGeneralTypeIssues) [Ln 28, Col 19]
"utils" is not a known member of module  Pylance(reportGeneralTypeIssues) [Ln 29, Col 19]
"utils" is not a known member of module  Pylance(reportGeneralTypeIssues) [Ln 30, Col 19]
"utils" is not a known member of module  Pylance(reportGeneralTypeIssues) [Ln 31, Col 19]
"utils" is not a known member of module  Pylance(reportGeneralTypeIssues) [Ln 32, Col 19]
```

At the moment I'm thinking the following should be the public interface:

```python
import pydantic_argparse

# Package Root
pydantic_argparse.ArgumentParser
pydantic_argparse.__title__
pydantic_argparse.__description__
pydantic_argparse.__version__
pydantic_argparse.__author__
pydantic_argparse.__license__

# Argparse Subpackage
pydantic_argparse.argparse
pydantic_argparse.argparse.ArgumentParser
```

...and everything else should be considered private for now.

I will merge #23 and follow-up with some extra `__all__` stuff to implement the above public interface.

Thanks!

**Comment by SupImDos** [comment]
Fixed in #23 and #24

**Comment by SupImDos** [comment]
Released in [`v0.6.1`](https://github.com/SupImDos/pydantic-argparse/releases/tag/v0.6.0) ([PyPI](https://pypi.org/project/pydantic-argparse/0.6.1/))
